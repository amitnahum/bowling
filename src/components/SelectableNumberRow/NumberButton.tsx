import React from 'react';

interface NumberButtonArguments {
    number: number,
    key: number,
    isSelected: boolean,
    onSelect: React.MouseEventHandler,
}

export function NumberButton({number, isSelected, key, onSelect}: NumberButtonArguments) {
    let classes = ["button"]
    if (isSelected) {
        classes.push('selected')
    }
    return (
        <div className={classes.join(' ')}  onClick={onSelect} key={key}>
            {number}
        </div>
    );
}
