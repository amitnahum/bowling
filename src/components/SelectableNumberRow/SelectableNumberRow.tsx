import React from 'react';
import { NumberButton } from "./NumberButton";


interface SelectableNumberRowArguments {
    numbers: number[],
    selected: number,
    onSelect: Function
}

/**
 * a row of numbers that allow for the selection of a number from the row
 * @param numbersLeft : number[]
 * @param selected : number
 * @param onSelect : Function
 * @constructor
 */
export function SelectableNumberRow({numbers, selected, onSelect}: SelectableNumberRowArguments) {
    return (
        <div style={{display: "flex"}}>
            {
                // iterate over the numbers we have left and render a NumberButton
                numbers.map((number: number, i: number) => NumberButton({
                    number,
                    isSelected: i === selected,
                    key: i,
                    onSelect: () => onSelect(i)
                }))
            }
        </div>
    );
}
