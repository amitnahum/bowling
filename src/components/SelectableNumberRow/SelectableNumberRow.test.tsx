import React from 'react';
import { SelectableNumberRow } from "./SelectableNumberRow";
import renderer from "react-test-renderer";

describe("Selectable Number Row Unit Tests", function () {
    it('renders buttons with value successfully ', function () {
        const selectableNumberRow = renderer.create(<SelectableNumberRow numbers={[4,3,2,1]} selected={1} onSelect={()=>{}}/>).toJSON();
        expect(selectableNumberRow).toMatchSnapshot();
    })
    it('renders buttons with value successfully #2', function () {
        const selectableNumberRow = renderer.create(<SelectableNumberRow numbers={[4]} selected={1} onSelect={()=>{}}/>).toJSON();
        expect(selectableNumberRow).toMatchSnapshot();
    })
    it('renders buttons with value successfully #3', function () {
        const selectableNumberRow = renderer.create(<SelectableNumberRow numbers={[4,2,1]} selected={2} onSelect={()=>{}}/>).toJSON();
        expect(selectableNumberRow).toMatchSnapshot();
    })

});
