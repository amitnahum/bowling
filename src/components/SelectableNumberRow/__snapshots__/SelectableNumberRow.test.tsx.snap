// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Selectable Number Row Unit Tests renders buttons with value successfully  1`] = `
<div
  style={
    Object {
      "display": "flex",
    }
  }
>
  <div
    className="button"
    onClick={[Function]}
  >
    4
  </div>
  <div
    className="button selected"
    onClick={[Function]}
  >
    3
  </div>
  <div
    className="button"
    onClick={[Function]}
  >
    2
  </div>
  <div
    className="button"
    onClick={[Function]}
  >
    1
  </div>
</div>
`;

exports[`Selectable Number Row Unit Tests renders buttons with value successfully #2 1`] = `
<div
  style={
    Object {
      "display": "flex",
    }
  }
>
  <div
    className="button"
    onClick={[Function]}
  >
    4
  </div>
</div>
`;

exports[`Selectable Number Row Unit Tests renders buttons with value successfully #3 1`] = `
<div
  style={
    Object {
      "display": "flex",
    }
  }
>
  <div
    className="button"
    onClick={[Function]}
  >
    4
  </div>
  <div
    className="button"
    onClick={[Function]}
  >
    2
  </div>
  <div
    className="button selected"
    onClick={[Function]}
  >
    1
  </div>
</div>
`;
